src/event_handler.py

```diff
@@ -37,7 +39,11 @@ def store_queued_events(_, __):
         try:
             decrypted_message = decrypt_message(message['Body'], decryption_key)
             event = event_from_json(decrypted_message)
-            write_to_audit_database(event, db_connection)
+            write_audit_event_to_audit_database(event, db_connection)
+            if event.event_type == 'session_event' and 'session_event_type' in event.details and event.details['session_event_type'] == 'idp_authn_succeeded':
```

Comment:
I think you can use `event.details.get('session_event_type')` instead of checking `in` beforehand - https://www.tutorialspoint.com/python/dictionary_get.htm

Reply:
Thanks. I have updated it.